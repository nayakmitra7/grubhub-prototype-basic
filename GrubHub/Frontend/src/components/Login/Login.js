import React, {Component} from 'react';
import '../../App.css';
import axios from 'axios';
import cookie from 'react-cookies';
import {Redirect} from 'react-router';

class Login extends Component{
    constructor(props){
        super(props);
        this.state = {
            username : "",
            password : "",
            authFlag : false,
            errorMessage : []
        }
        this.usernameChangeHandler = this.usernameChangeHandler.bind(this);
        this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
        this.submitLogin = this.submitLogin.bind(this);
    }
    componentWillMount(){
        this.setState({
            authFlag : false
        })
    }
    usernameChangeHandler = (e) => {
        this.setState({
            username : e.target.value
        })
    }
    passwordChangeHandler = (e) => {
        this.setState({
            password : e.target.value
        })
    }
    submitLogin = (e) => {
        var headers = new Headers();
        e.preventDefault();
        const data = {
            username : this.state.username,
            password : this.state.password
        }
        axios.defaults.withCredentials = true;
        axios.post('http://localhost:3001/login',data)
            .then(response => {
                console.log("Status Code : ",response.status);
                if(response.status === 200){
                    sessionStorage.setItem("username",this.state.username);
                    this.setState({
                        authFlag : true
                    })
                }
                else if(response.status === 201){
                    console.log(response.data);
                    this.setState({
                        authFlag : false,
                        errorMessage : response.data
                    })
                }
            });
    }

    render(){
      var redirectVar="";
        if(cookie.load('cookie')){
            redirectVar = <Redirect to= "/HomePage"/>
        }
        let displayMessage = null;
        if(this.state.authFlag==false){
            displayMessage = ( this.state.errorMessage.map( (error) =>{
                return (<li class="li alert-danger">{error.msg}</li>)
            }))
        }
        return(
            <div>
                {redirectVar}
                {displayMessage} 
            <div class = "backgroundImg">
            <div class="container">            
                <div class="login-form">
                    <div class="main-div">
                        <div class="panel">
                            <h2 class="heading"> Sign in with your Grubhub account</h2>
                        </div>
                            <div class="form-group">
                                <p >Email</p>
                                <input onChange = {this.usernameChangeHandler} type="text" class="form-control" name="username" />
                            </div>
                            <div class="form-group">
                            <p>Password</p>
                                <input onChange = {this.passwordChangeHandler} type="password" class="form-control" name="password" />
                            </div>
                            <br></br>
                            <button onClick = {this.submitLogin} class="btn btn-danger">Sign-In</button>                
                            <br></br><br></br><br></br><br></br>
                            <a href="/signup" class="createAcc">Create your account</a>
                    </div>
                </div>
            </div>
            </div>
            </div>
        )
    }
}
//export Login Component
export default Login;