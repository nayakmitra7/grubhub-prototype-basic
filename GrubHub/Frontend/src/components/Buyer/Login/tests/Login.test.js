import React from 'react';
import Login from '../Login';
import renderer from 'react-test-renderer';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('render correctly Login component', () => {  
  const LoginInputComponent = renderer.create(<Login />).toJSON();
  expect(LoginInputComponent).toMatchSnapshot();
})


