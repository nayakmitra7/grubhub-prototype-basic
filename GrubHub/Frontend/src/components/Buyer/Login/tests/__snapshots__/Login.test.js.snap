// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Login component 1`] = `
<div>
  
  <div
    class="backgroundImg"
  >
    <div
      class="container"
    >
      <div
        class="login-form"
      >
        <div
          class="main-div"
        >
          <div
            class="panel"
          >
            <h2
              class="heading"
            >
               Sign in with your Grubhub account
            </h2>
          </div>
          <div
            class="form-group"
          >
            <p>
              Email
            </p>
            <input
              class="form-control"
              name="username"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            class="form-group"
          >
            <p>
              Password
            </p>
            <input
              class="form-control"
              name="password"
              onChange={[Function]}
              type="password"
            />
          </div>
          <br />
          <button
            class="btn btn-danger"
            onClick={[Function]}
          >
            Sign-In
          </button>
          <br />
          <br />
          <br />
          <br />
          <a
            class="createAcc"
            href="/signup"
          >
            Create your account
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
